lib/jenkins_pipeline_builder/pull_request_generator.rb

```diff
@@ -19,149 +19,53 @@
-          job = i[:result]
-          job = Job.new job
-          success, payload = job.create_or_update
-          errors[job.name] = payload unless success
-        end
-      end
-      errors
+    def delete_closed_prs
+      return if JenkinsPipelineBuilder.debug
+      jobs_to_delete = JenkinsPipelineBuilder.client.job.list "#{application_name}-PR.*"
+      open_prs.each { |n| jobs_to_delete.reject! { |j| j.start_with? "#{application_name}-PR#{n}" } }
+      jobs_to_delete.each { |j| JenkinsPipelineBuilder.client.job.delete j }
     end
 
-    def collection
-      generator.job_collection.collection
-    end
-
-    def filter_jobs
-      jobs = {}
-      pull_jobs = pull_generator[:value][:jobs] || []
-      pull_jobs.each do |job|
-        if job.is_a? String
-          jobs[job.to_s] = collection[job.to_s]
-        else
-          jobs[job.keys.first.to_s] = collection[job.keys.first.to_s]
-        end
-      end
-      fail 'No jobs found for pull request' if jobs.empty?
-      jobs
-    end
-
-    def compile_generator
-      defaults = generator.job_collection.defaults
-      settings = defaults.nil? ? {} : defaults[:value] || {}
-      compiler = Compiler.new generator
-      settings = compiler.get_settings_bag(project, settings)
-      success, payload = generator.resolve_job_by_name(pull_generator[:name], settings)
-      return payload if success
-
-      @errors[@pull_generator[:name]] = payload
-      false
-    end
-
-    def find
-      project_jobs = project[:value][:jobs] || []
-      pull_job = nil
-      project_jobs.each do |job|
-        job = job.keys.first if job.is_a? Hash
-        job = collection[job.to_s]
+    private
 
-        pull_job = job if job[:value][:job_type] == 'pull_request_generator'
-      end
-      fail NotFound, 'No jobs of type pull_request_generator found' unless pull_job
-      pull_job
+    def overrides(pr)
+      git_version = JenkinsPipelineBuilder.registry.registry[:job][:scm_params].installed_version
+      override = {
+        scm_branch: "origin/pr/#{pr}/head",
+        scm_params: {
+          refspec: "refs/pull/#{pr}/head:refs/remotes/origin/pr/#{pr}/head"
+        }
+      }
+      override[:scm_params][:changelog_to_branch] = {
+        remote: 'origin', branch: "pr/#{pr}/head"
+      } if git_version >= Gem::Version.new(2.0)
+      override
     end
 
-    # Check for Github Pull Requests
-    #
-    # args[:git_url] URL to the github main page ex. https://www.github.com/
-    # args[:git_repo] Name of repo only, not url  ex. jenkins_pipeline_builder
-    # args[:git_org] The Orig user ex. constantcontact
-    # @return = array of pull request numbers
-    def check_for_pull(args)
-      fail 'Please specify all arguments' unless args[:git_url] && args[:git_org] && args[:git_repo]
+    def active_prs(git_url: nil, git_org: nil, git_repo: nil)
+      (git_url && git_org && git_repo) || fail('Please set github_site, git_org and git_repo_name in your project.')
```

Comment:
I think I'd rather move this checking into the initialize and set these as instance vars with attr_reader as well.